src/codegen/mod.rs

```diff
@@ -1383,9 +1791,10 @@ fn gen_ty(ty: &desc::TyKind, mutbl: desc::Mutability) -> cu::Ty {
     let cu_ty = match ty {
         Ident(ident) => cu::Ty::Ident(ident.name.clone()),
         Fn(_, _, _, _) => unimplemented!("needed?"),
-        Data(d::Scalar(s)) => match s {
+        Data(d::Scalar(s)) | Data(d::Atomic(s)) => match s {
```

Comment:
`cu::Atomic` is not generated here. This suggests to me that the `Atomic` variant in` `cu_ast` is not needed.

Reply:
now this gets generated here